groupcache.go

```diff
@@ -416,57 +436,62 @@ func (g *Group) store(ctx Context, key string, data []byte, ttl time.Duration) (
 		}
 		g.Stats.LocalStores.Add(1)
 		value := ByteView{b: data}
-		g.populateCache(key, value, &g.mainCache)
+		g.populateCache(key, payload{value: value, ttl: ttl}, &g.mainCache)
 		return nil, nil
 	})
 	return
 }
 
-func (g *Group) putLocally(ctx Context, key string, data []byte, ttl time.Duration) error {
+func (g *Group) putLocally(ctx Context, key string, data []byte, ttl *time.Time) error {
 	return g.putter.Put(ctx, key, data, ttl)
 }
 
-func (g *Group) putFromPeer(ctx Context, peer ProtoPeer, key string, data []byte, ttl time.Duration) error {
+func (g *Group) putFromPeer(ctx Context, peer ProtoPeer, key string, data []byte, ttl *time.Time) error {
+	ttlProto, err := ptypes.TimestampProto(*ttl)
+	if err != nil {
+		return err
+	}
 	req := &pb.PutRequest{
 		Group: g.name,
 		Key:   key,
 		Value: data,
-		Ttl:   ptypes.DurationProto(ttl),
+		Ttl:   ttlProto,
 	}
 	res := &pb.PutResponse{}
-	err := peer.Put(ctx, req, res)
+	err = peer.Put(ctx, req, res)
 	if err != nil {
 		return err
 	}
 	value := ByteView{b: data}
```

Comment:
big nit: this could be pulled into the `if rand.Intn()...` block too

Reply:
I try to avoid refactors like that in PR for Features, but I totally agree. I was debating it myself. Done.

Reply:
You are referring to value := ByteView{b: data}, correct?

Reply:
yup! agreed in general, but it's a small enough change that I think it's ok

Reply:
Totally agree, I honestly thought about doing it for like 5 minutes. Glad you mentioned this.